6_nodejs/week3/app.js

```diff
@@ -0,0 +1,53 @@
+import knex from "knex";
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "pass",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from "express";
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    if (orderBy === "first_name") {
+      query = query.orderBy(orderBy);
+    } else if (orderBy === "last_name") {
+      query = query.orderBy(orderBy, "desc");
+    } else {
+      res.send("invalid sort");
+    }
```

Comment:
Well done on the video of the injection attack. 

I see you have fixed the issue by making sure the parameter value is allowed. Good job. 💪  One note is that you did this at the expense of functionality: currently there is no way for the user to determine the ordering direction. First names are always ordered ascending, last names are always ordered descending. You could fix this by implementing another parameter, for example `sortDir`.